components/style/shared_lock.rs

```diff
@@ -0,0 +1,203 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+//! Different objects protected by the same lock
+
+use parking_lot::RwLock;
+use std::cell::UnsafeCell;
+use std::fmt;
+use std::ops::Deref;
+use std::sync::Arc;
+
+/// A shared read/write lock that can protect multiple objects.
```

Comment:
I originally intended this module to be a general purpose library that could be its own crate, so I kept the doc-comment generic. (Later I added `ToCssWithGuard` and `ReadGuards` there because that seemed the appropriate place for them, but they would remain in the `style` crate.)